docs/getting-started/first-steps.md

```
@@ -0,0 +1,201 @@
+
+# State objects are immutable
+prefs = UserPreferences()
+# prefs.theme = "dark"  # This would raise an error!
+
+# Instead, create updated instances through copies
+dark_prefs: UserPreferences = prefs.updated(theme="dark")
+print(f"Original: {prefs.theme}, Updated: {dark_prefs.theme}")
+```
+
+**What's happening here:**
+- **Automatic Immutability**: `State` base class prevents modification after creation using `__setattr__` blocking
```

markdownlint-cli2 (0.17.2)

Lists should be surrounded by blank lines
null

(MD032, blanks-around-lists)